spec/models/data_update_script_spec.rb

```diff
@@ -63,6 +63,41 @@
     end
   end
 
+  describe ".scripts_to_run?" do
+    let(:test_directory) { Rails.root.join("spec/support/fixtures/data_update_scripts") }
+
+    before { stub_const "#{described_class}::DIRECTORY", test_directory }
+
+    it "returns true for a new set of files" do
+      expect(described_class.scripts_to_run?).to be(true)
+    end
+
+    it "returns true if there is a script on disk but not in the DB" do
+      create(:data_update_script, file_name: "not_the_one_on_disk")
+      expect(described_class.scripts_to_run?).to be(true)
+    end
+
+    it "returns true if there is an enqueued script" do
+      create(:data_update_script, status: :enqueued)
+      expect(described_class.scripts_to_run?).to be(true)
+    end
+
+    it "returns false if there are only working scripts" do
+      create(:data_update_script, status: :working)
+      expect(described_class.scripts_to_run?).to be(false)
+    end
+
+    it "returns false if there are only succeeded scripts" do
+      create(:data_update_script, status: :succeeded)
+      expect(described_class.scripts_to_run?).to be(false)
+    end
+
+    it "returns false if there are only failed scripts" do
+      create(:data_update_script, status: :failed)
+      expect(described_class.scripts_to_run?).to be(false)
```

Comment:
Hm, this behavior feels a little odd to me. I'm thinking of running scripts similar to how we'd run migrations. If a migration couldn't be run, and I run `rake db:migrate:status`, I would still expect that migration to be marked `down` rather than `up`.

Is there a reason we want to return `false` explicitly on failure here that I'm missing?

Reply:
You read our minds, I've just had this conversation in private with @maestromac :D 

The thing is, I didn't want to change the existing behavior, and the reason for that is:

https://github.com/thepracticaldev/dev.to/blob/e59e523ace955da272013ebd8e8c5f10e4abb098/app/workers/data_update_worker.rb#L5-L12

which calls:

https://github.com/thepracticaldev/dev.to/blob/e59e523ace955da272013ebd8e8c5f10e4abb098/app/models/data_update_script.rb#L25-L27

Currently we only run new scripts, there's no mechanism to re-run failed scripts.

I think @mstruve designed it as such in the beginning. My opinion is that before allowing the devs to re-run a failed script in dev we need to be absolutely sure that:

- that failed script can be re-run (I'm sure not all scripts will be idempotent in the future)
- we store the failure message (have a PR yet to be written about this)

In the meantime I think we should keep it as-is, we can easily change this behavior in a separate PR if we come to a consensus

Reply:
Ok, that's fair; totally didn't realize we didn't have the mechanism to re-run failed scripts yet! 🙈 In that case, I think maintaining feature parity here makes sense.

(Please excuse my ignorance here, I'm still kind of new to this section of the codebase!)

Reply:
I am learning about `DataUpdateScript` as I go as well :D

Reply:
Ideally, in the future we get a little Dashboard together for people to rerun their failed scripts from admin. All scripts should be idempotent as it states in the comments in the template. It's up to us to ensure that continues to be the case as we review them.

Reply:
love the idea of being able to re-run scripts in the admin dashboard! seems like a good cool down project at some point 🙃

Reply:
I'm new to this too. Thanks for clarifying @rhymes @mstruve !